GameSentenceMiner/util/db.py

```diff
@@ -615,6 +615,107 @@ def get_lines_filtered_by_timestamp(cls, start: Optional[float] = None, end: Opt
         clean_columns = ['line_text'] if for_stats else []
         return [cls.from_row(row, clean_columns=clean_columns) for row in rows]
 
+
+class GoalsTable(SQLiteDBTable):
+    """
+    Table for tracking daily goal completions and streaks.
+    One entry per day when user completes their dailies.
+    """
+    _table = 'goals'
+    _fields = ['date', 'current_goals', 'goals_settings', 'last_updated']
```

Comment:
current_goals = user generate goals
goals_settings = easy days etc
last_updated == easier to sync later by just keeping last updated data
we work out streaks based on dates, everytime a user clicks "complete dailies" we write their settings and current_goals to db with the date

we store this in the db because:
* local storage is weak and user will be very upset if their data is lost
* Unlike other stats, this one is created by the user
* We can possibly generate a trophy cabinet down the road of all the goals the user has achieved